dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/TokenAwareSelection.java

```diff
@@ -79,30 +77,46 @@ public Long get(HostToken x) {
     }
 
     /**
-     * If a hashtag is provided by Dynomite then we use that to create the key to hash.
+     * Identifying the proper pool for the operation. A couple of things that may affect the decision
+     * (a) hashtags: In this case we will construct the key by decomposing from the hashtag
+     * (b) type of key: string keys vs binary keys. 
+     * In binary keys hashtags do not really matter.
      */
     @Override
     public HostConnectionPool<CL> getPoolForOperation(BaseOperation<CL, ?> op, String hashtag) throws NoAvailableHostsException {
```

Comment:
I am no expert, but I am wondering if its a good idea to split the two implementations into something like:
StringBaseOperation &
BinaryBaseOperation? so there is not if else anywhere?

Compression is implemented inside them may be.

Reply:
are you planning to do this? or should I merge?

Reply:
I looked into it and it becomes very intrusive and it would require more code. We will need `StringBaseOperation`, `BinaryBaseOperation` and each one of them for the multikeys, then go to each command and use the corresponding one. Currently, the overhead of having a null byte array or a null string is very minimal. I suggest if we go in this direction to do it in a separate PR. Let us make sure that after releasing an RC, clients see a benefit.